src/test/app/SetTrust_test.cpp

```diff
@@ -0,0 +1,79 @@
+//------------------------------------------------------------------------------
+/*
+This file is part of rippled: https://github.com/ripple/rippled
+Copyright (c) 2016 Ripple Labs Inc.
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose  with  or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/test/jtx.h>
+
+namespace ripple {
+
+namespace test {
+
+class SetTrust_test : public beast::unit_test::suite
+{
+public:
+    void testFreeTrustlines()
+    {
+        testcase("Allow 2 free trust lines before requiring reserve");
+
+        using namespace jtx;
+        Env env(*this);
+
+        auto const gwA = Account{ "gatewayA" };
+        auto const gwB = Account{ "gatewayB" };
+        auto const gwC = Account{ "gatewayC" };
+        auto const alice = Account{ "alice" };
+        
```

Comment:
Nit pick: extra space here.

I recommend you use a utility to automatically remove extra spaces when you save. If you use VS, this one is free and works well: https://visualstudiogallery.msdn.microsoft.com/a204e29b-1778-4dae-affd-209bea658a59